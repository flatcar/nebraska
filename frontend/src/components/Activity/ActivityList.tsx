import MuiList from '@mui/material/List';
import Typography from '@mui/material/Typography';
import makeStyles from '@mui/styles/makeStyles';
import React from 'react';

import { Activity } from '../../api/apiDataTypes';
import { makeLocaleTime } from '../../i18n/dateTime';
import ActivityItem from './ActivityItem';

const useStyles = makeStyles({
  listTitle: {
    fontSize: '1em',
  },
});

export interface ActivityListProps {
  entries?: Activity[];
  timestamp: string;
}

export default function ActivityList(props: ActivityListProps) {
  const classes = useStyles();
  const entries = props.entries ? props.entries : [];

  return (
    <React.Fragment>
      <Typography className={classes.listTitle}>
        {makeLocaleTime(props.timestamp, {
          showTime: false,
          dateFormat: { weekday: 'long', month: 'long', day: 'numeric', year: 'numeric' },
        })}
      </Typography>
      <MuiList>
        {entries.map((entry: Activity) => (
          <ActivityItem key={entry.id} entry={entry} />
        ))}
      </MuiList>
    </React.Fragment>
  );
}
