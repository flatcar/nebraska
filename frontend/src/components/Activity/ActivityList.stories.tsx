import { Meta, StoryFn } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

import ActivityList, { ActivityListProps } from './ActivityList';

export default {
  title: 'activity/ActivityList',
} as Meta;

const Template: StoryFn<ActivityListProps> = args => (
  <MemoryRouter>
    <ActivityList {...args} />
  </MemoryRouter>
);

export const Empty = {
  render: Template,

  args: {
    timestamp: 'Wed, 13 May 2020 14:56:03 GMT',
  },
};

export const BetaList = {
  render: Template,

  args: {
    timestamp: 'Wed, 13 May 2020 14:56:03 GMT',
    entries: [
      {
        id: 1,
        app_id: '',
        group_id: '',
        created_ts: '2020-05-13T20:26:03.837688+05:30',
        class: 6,
        severity: 2,
        version: '0.0.0',
        application_name: 'ABC',
        group_name: null,
        channel_name: 'beta',
        instance_id: null,
      },
      {
        id: 2,
        app_id: '',
        group_id: '',
        created_ts: '2020-05-13T20:25:52.589886+05:30',
        class: 6,
        severity: 2,
        version: '0.0.0',
        application_name: 'DEF',
        group_name: null,
        channel_name: 'beta',
        instance_id: null,
      },
    ],
  },
};
