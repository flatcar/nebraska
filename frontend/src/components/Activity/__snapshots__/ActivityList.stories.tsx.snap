// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity/ActivityList BetaList smoke-test 1`] = `
<p class="MuiTypography-root MuiTypography-body1 css-1c4wxug-MuiTypography-root">
  Wednesday, May 13, 2020
</p>
<ul class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root">
  <li class="MuiListItem-root MuiListItem-padding MuiListItem-alignItemsFlexStart css-kd2s28-MuiListItem-root">
    <div class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-tqvfrr-MuiGrid-root">
      <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-10 css-r13beu-MuiGrid-root">
        <div class="MuiBox-root css-geek62">
          <div class="MuiBox-root css-12z0wuy">
            <svg xmlns="http://www.w3.org/2000/svg"
                 xmlns:xlink="http://www.w3.org/1999/xlink"
                 aria-hidden="true"
                 role="img"
                 width="30px"
                 height="30px"
                 viewbox="0 0 24 24"
                 style="color: rgb(0, 211, 255);"
            >
              <path fill="currentColor"
                    d="M11 15h2v2h-2v-2m0-8h2v6h-2V7m1-5C6.47 2 2 6.5 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2m0 18a8 8 0 0 1-8-8a8 8 0 0 1 8-8a8 8 0 0 1 8 8a8 8 0 0 1-8 8Z"
              >
              </path>
            </svg>
          </div>
          <div class="MuiBox-root css-0">
            <p class="MuiTypography-root MuiTypography-body1 css-10o1zsr-MuiTypography-root">
              ABC
            </p>
          </div>
        </div>
      </div>
      <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-2 css-15bfe7d-MuiGrid-root">
        <p class="MuiTypography-root MuiTypography-body1 css-1nwggdl-MuiTypography-root">
          02:56 PM
        </p>
      </div>
      <div class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root">
        Channel beta is now pointing to version 0.0.0
      </div>
    </div>
  </li>
  <li class="MuiListItem-root MuiListItem-padding MuiListItem-alignItemsFlexStart css-kd2s28-MuiListItem-root">
    <div class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-tqvfrr-MuiGrid-root">
      <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-10 css-r13beu-MuiGrid-root">
        <div class="MuiBox-root css-geek62">
          <div class="MuiBox-root css-12z0wuy">
            <svg xmlns="http://www.w3.org/2000/svg"
                 xmlns:xlink="http://www.w3.org/1999/xlink"
                 aria-hidden="true"
                 role="img"
                 width="30px"
                 height="30px"
                 viewbox="0 0 24 24"
                 style="color: rgb(0, 211, 255);"
            >
              <path fill="currentColor"
                    d="M11 15h2v2h-2v-2m0-8h2v6h-2V7m1-5C6.47 2 2 6.5 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2m0 18a8 8 0 0 1-8-8a8 8 0 0 1 8-8a8 8 0 0 1 8 8a8 8 0 0 1-8 8Z"
              >
              </path>
            </svg>
          </div>
          <div class="MuiBox-root css-0">
            <p class="MuiTypography-root MuiTypography-body1 css-10o1zsr-MuiTypography-root">
              DEF
            </p>
          </div>
        </div>
      </div>
      <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-2 css-15bfe7d-MuiGrid-root">
        <p class="MuiTypography-root MuiTypography-body1 css-1nwggdl-MuiTypography-root">
          02:55 PM
        </p>
      </div>
      <div class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root">
        Channel beta is now pointing to version 0.0.0
      </div>
    </div>
  </li>
</ul>
`;

exports[`activity/ActivityList Empty smoke-test 1`] = `
<p class="MuiTypography-root MuiTypography-body1 css-1c4wxug-MuiTypography-root">
  Wednesday, May 13, 2020
</p>
<ul class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root">
</ul>
`;
